### PR TITLE
etcdserver: refactor etcdserver initialization code

### PR DESCRIPTION
**Pull request by heyitsanthony**
moved the raft node start logic to separate functions

### CODE REVIEW

File: etcdserver/raft.go [hunk 1]
```diff
@@ -85,6 +85,15 @@ type apply struct {
 	raftDone <-chan struct{} // rx {} after raft has persisted messages
 }
 
+// activeRaft represents an activated raft instance via raft.{Res,S}tartNode
+type activeRaft struct {
```
Comment (xiang90) [inline etcdserver/raft.go]: I feel we need a better name for this struct. "activeRaft, raftNode, raft.Node": we have too many rafts in etcdserver.

File: etcdserver/server.go [hunk 2]
```diff
@@ -223,137 +214,55 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
...
-				return nil, fmt.Errorf("discovery cluster %s has duplicate url", urlsmap)
-			}
-			if cl, err = newClusterFromURLsMap(cfg.InitialClusterToken, urlsmap); err != nil {
-				return nil, err
-			}
-		}
-		cl.SetStore(st)
-		cfg.PrintWithInitial()
-		id, n, s, w = startNode(cfg, cl, cl.MemberIDs())
+		ar, err = joinBootstrap(cfg, prt)
```
Comment (xiang90) [inline etcdserver/server.go]: probably just "bootstrap" instead of "joinBootstrap"?

File: etcdserver/server.go [hunk 3]
```diff
@@ -223,137 +214,55 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
...
-		}
-		cfg.Print()
-		if !cfg.ForceNewCluster {
-			id, cl, n, s, w = restartNode(cfg, snapshot)
-		} else {
-			id, cl, n, s, w = restartAsStandaloneNode(cfg, snapshot)
-		}
-		cl.SetStore(st)
-		cl.Recover()
+		ar, err = joinRecover(cfg, prt)
```
Comment (xiang90) [inline etcdserver/server.go]: probably just recover

File: etcdserver/raft.go (hunk 1 above)
Comment (heyitsanthony) [inline etcdserver/raft.go]: `activeNode`? The functions returning the new structure already have a `Node` in the suffix

Comment (xiang90) [inline etcdserver/raft.go]: give me a little bit more time to think about the naming... it is so hard.

**Comment by xiang90** [comment]
@heyitsanthony Can we close this for now? We still should clean this up. But let's plan this after 3.1?

**Comment by heyitsanthony** [comment]
sure, closing

**Pull request closed without merging**